src/pages/home/report/ReportActionsList.tsx

```diff
@@ -190,9 +190,9 @@ function ReportActionsList({
                         ReportActionsUtils.isDeletedParentAction(reportAction) ||
                         reportAction.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE ||
                         reportAction.errors) &&
-                    ReportActionsUtils.shouldReportActionBeVisible(reportAction, reportAction.reportActionID),
+                    ReportActionsUtils.shouldReportActionBeVisible(reportAction, reportAction.reportActionID, ReportUtils.canUserPerformWriteAction(report)),
             ),
-        [sortedReportActions, isOffline],
+        [sortedReportActions, isOffline, report],
```

Comment:
Having the report here was retriggering the memo block multiple times, which caused this blocker https://github.com/Expensify/App/issues/52891